api/python/quilt3/data_transfer.py

```diff
@@ -269,14 +266,17 @@ def _upload_file(ctx, size, src_path, dest_bucket, dest_key):
                 Body=fd,
                 Bucket=dest_bucket,
                 Key=dest_key,
+                ChecksumAlgorithm='SHA256',
             )
 
         version_id = resp.get('VersionId')  # Absent in unversioned buckets.
-        ctx.done(PhysicalKey(dest_bucket, dest_key, version_id))
+        sha256 = resp['ChecksumSHA256']
+        ctx.done(PhysicalKey(dest_bucket, dest_key, version_id, sha256))
```

Comment:
Does the now additional passing of `sha256` guarantee that this serializes with the new hash type?

Reply:
When this physical key serializes in a manifest it will have a different type, right (not `"type": "sha256"`)? I'm just wondering how the manifest serializer knows that.

Reply:
Oh, serialization only looks at `PackageEntry`, not at `PhysicalKey`. The type gets set here: https://github.com/quiltdata/quilt/pull/2782/files#diff-652f7ab61c8420f5d72f57c54f12337d64a53343502a81d912026e0831bd53e0R1464

Reply:
Is this resolved?